ci/tasks/run-lifecycle.sh

```diff
@@ -35,9 +35,11 @@ if [ "$BOSH_VSPHERE_CPI_NSXT_HOST" != "null" ]; then
   BOSH_NSXT_CA_CERT_FILE="$(realpath "${PWD}/nsxt-manager-cert.pem")"
   # To get the cert from nsxt-manager, we run openssl on the jump box, and then pipe that result into a local openssl command that reformats it into PEM
   openssl s_client -showcerts -connect "${BOSH_VSPHERE_CPI_NSXT_HOST}":443 </dev/null 2>/dev/null | openssl x509 -outform PEM > $BOSH_NSXT_CA_CERT_FILE
-  # The certificate's subject name is nsxt-manager so SSL validation fails when using the IP address
-  echo "${BOSH_VSPHERE_CPI_NSXT_HOST} nsxt-manager" >> /etc/hosts
-  export BOSH_VSPHERE_CPI_NSXT_HOST="https://nsxt-manager"
+  # The certificate's SAN contains the host name, so extract it because SSL validation fails when using the IP address
+  # NOTE: Don't hard code the name as it is not guaranteed to be "nsxt-manager"
+  BOSH_NSXT_CERT_HOST_NAME="$(openssl x509 -noout -text -in nsxt-manager-cert.pem | awk '/X509v3 Subject Alternative Name/ {getline;gsub(/ /, "", $0); print}' | tr -d "DNS:" | sed 's/^\*./nsx-mgr\./')"
```

Comment:
> sed 's/^\*./nsx-mgr\./

the `.` on the search string is not escaped, so it will match anything. Perhaps escape just to be safe?

in the replacement string, you've escaped the dot, `\.`. We don't believe that escape is necessary here.

To remove some of the text process, you could use `openssl x509 -noout -in nsxt-manager-cert.pem -ext 'subjectAltName'`. The output of that command is something like:
```
X509v3 Subject Alternative Name:
    DNS:github.com, DNS:www.github.com
```

Reply:
Added the suggestions of escaping the . in the search string and removing the escape in the replacement string.

Adding `-ext 'subjectAltName` is resulting in the new name being returned twice, so I'm leaving out that suggestion for now.